main.go

```diff
@@ -19,105 +18,119 @@ type FileInfo map[string]time.Time
 var watchedFiles = FileInfo{}
 
 func main() {
-	defaultPath, _ := os.Getwd()
+	defaultPath, err := os.Getwd()
+	if err != nil {
+		log.Fatal("could not get current directory::", err)
+	}
+
 	rootPath := flag.String("path", defaultPath, "the root path to be watched")
-	interval := flag.Int("interval", 5, "the interval (in seconds) for scanning files")
+	interval := flag.Duration("interval", 5*time.Second, "the interval (in seconds) for scanning files")
 	flag.Parse()
 
-	watchedFiles = walk(rootPath)
-	scheduleScanAt(rootPath, interval)
-}
+	if *interval < 0 {
+		log.Fatal("invalid interval, must be > 0:", *interval)
+	}
 
-func scheduleScanAt(rootPath *string, interval *int) {
-	runTicker := time.NewTicker(
-		time.Duration(time.Duration(*interval) * time.Second))
-	for {
-		select {
-		case <-runTicker.C:
-			scan(rootPath)
-		}
+	if err := os.Chdir(*rootPath); err != nil {
+		log.Fatal("could not change directory:", err)
+	}
+
+	watchedFiles = walk(rootPath)
+	for range time.NewTicker(*interval).C {
+		scan(rootPath)
 	}
 }
 
 func scan(rootPath *string) {
+	modifiedDirs := make(map[string]bool, 0)
```

Comment:
```suggestion
	modifiedDirs := []string{}
```

Reply:
You clearly did not understand the code.